### PR TITLE
docs: get oauth2_client api contract changes

### PR DESCRIPTION
**Pull request by AdithyaSSup**
changes made on get oauth2_client api contract  and added a flow diagram

### CODE REVIEW

File: api_spec.yaml [hunk 1]
```diff
@@ -708,28 +709,27 @@ paths:
...
-                      scopes:
-                        type: array
-                        items:
-                          $ref: '#/components/schemas/oauth2Scope'
-                      timeCreated:
-                        $ref: '#/components/schemas/timestampMs'
-                      timeUpdated:
-                        $ref: '#/components/schemas/timestampMs'
+                      clientInfo:
+                        $ref: '#/components/schemas/oauth2ClientInfo'
```
Comment (rishabhpoddar) [inline api_spec.yaml:721]: i don't think that we store the client secret in the database, but instead, we store the client secret hash. So the output of this API cannot have the clientSecret.

You can make the clientSecret field optional in the oauth2ClientInfo type

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 1 above)
Comment (AdithyaSSup) [reply api_spec.yaml:721]: changes made as below:.
  1. made `clientSecret` as optional in `Oauth2ClientSchema`
  2. made `clientSecreat` as required in create oauth2 Client api response

File: api_spec.yaml [hunk 2]
```diff
@@ -6854,7 +6858,6 @@ components:
       required:
         - name
         - clientId
-        - clientSecret
         - redirectUris
         - timeCreated
         - timeUpdated
```
Comment (rishabhpoddar) [inline api_spec.yaml:6863]: scopes is also required

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 2 above)
Comment (AdithyaSSup) [reply api_spec.yaml:6866]: changes  made

**Pull request merged**